qiskit_experiments/characterization/spectroscopy.py

```diff
@@ -0,0 +1,385 @@
+        amp_bounds: Optional[Tuple[float, float]] = None,
+        sigma_bounds: Optional[Tuple[float, float]] = None,
+        freq_bounds: Optional[Tuple[float, float]] = None,
+        offset_bounds: Optional[Tuple[float, float]] = None,
+        plot: bool = True,
+        ax: Optional["AxesSubplot"] = None,
+        **kwargs,
+    ) -> Tuple[AnalysisResult, None]:
+        """
+        Analyse a spectroscopy experiment by fitting the data to a Gaussian function.
+        The fit function is:
+
+        .. math::
+
+            a * exp(-(x-x0)**2/(2*sigma**2)) + b
+
+        Here, :math:`x` is the frequency. The analysis loops over the initial guesses
+        of the width parameter :math:`sigma`.
+
+        Args:
+            experiment_data: The experiment data to analyze.
+            data_processor: The data processor with which to process the data.
+            meas_level: The measurement level of the experiment data.
+            meas_return: Whether single-shot (the default) or average data is returned by the
+                experiment.
+            amp_guess: The amplitude of the Gaussian function, i.e. :math:`a`. If not
+                provided, this will default to the maximum absolute value of the ydata.
+            sigma_guesses: The guesses for the standard deviation of the Gaussian distribution.
+                If it is not given this will default to an array of ten
+                points linearly spaced between zero and the full width of the data.
+            freq_guess: A guess for the frequency of the peak :math:`x0`. If not provided
+                this guess will default to the location of the highest absolute data point.
+            offset_guess: A guess for the magnitude :math:`b` offset of the fit function.
+                If not provided, the initial guess defaults to the average of the ydata.
+            amp_bounds: Bounds on the amplitude of the Gaussian function as a tuple of
+                two floats. The default bounds are [0, 1.1*max(ydata)]
+            sigma_bounds: Bounds on the standard deviation of the Gaussian function as a tuple
+                of two floats. The default values are [0, frequency range].
+            freq_bounds: Bounds on the center frequency as a tuple of two floats. The default
+                values are 90% of the lower end of the frequency and 110% of the upper end of
+                the frequency.
+            offset_bounds: Bounds on the offset of the Gaussian function as a tuple of two floats.
+                The default values are the minimum and maximum of the ydata.
+            plot: If True generate a plot of fitted data.
+            ax: Optional, matplotlib axis to add plot to.
+            kwargs: Trailing unused function parameters.
+
+        Returns:
+            The analysis result with the estimated peak frequency.
+
+        Raises:
+            ValueError: If the measurement level is not supported.
+        """
+
```

Comment:
We could explore:
```suggestion
if data_processors is None:
    data_processor = ProcessorLibrary.get_default(measurement_level=meas_level, meas_outcome=meas_outcome)
```
where `ProcessorLibrary.get_default` could return either a `DataProcessor` or a `callable`.